### ISSUE TITLE
Support for Exif.NikonCb2b.WB_RGGBLevels

### ISSUE THREAD

**Issue by StyXman** [exiftool/exiftool#218]
Last night I took a series of 500+ photos for a time lapse. It's a sunset TL, so ISO and aperture were fixed; I let the camera control the exposure time. WB is also fixed.

Going through the images, I see slight variations in brightness. Looking at the calculated exposure times, I see that three consecutive photos (base - darker - base) have the same ET.

I tried to see if there were more info in the EXIF, so I run `exiftool` on a base and the darker picture and see no particular differences:

```
$ exiftool 2023-08-08T19.18.31.jpg > exiftool-31.txt
$ exiftool 2023-08-08T19.18.41.jpg > exiftool-41.txt
$ diff -u exiftool-31.txt exiftool-41.txt
```

The I tried an exif lib for Python (sorry!) and got (among many others) this:

```python
python3 -c 'import pprint, sys, gi
gi.require_version("GExiv2", "0.10")
from gi.repository import GExiv2
md = GExiv2.Metadata(sys.argv[1])
pprint.pprint({ key: md[key] for key in md.get_tags() })' 2023-08-08T19.18.31.jpg > gexiv-31.txt
```

(similar for `2023-08-08T19.18.41.jpg`)

```diff
$ diff -u gexiv-31.txt gexiv-41.txt
[...]
- 'Exif.NikonCb2b.0x0002': '45374 4007 7124 47327 5291 7697 52953 12233 47486 '
[...]
- 'Exif.NikonCb2b.0x008e': '11308 3132 7180',
- 'Exif.NikonCb2b.0x0095': '42196 40092 58227 36252 40348 54692 50987 62532 '
[...]
- 'Exif.NikonCb2b.WB_RGGBLevels': '27756 21604 17652 54468',
+ 'Exif.NikonCb2b.WB_RGGBLevels': '9468 54492 50252 31812',
[...]
  'Exif.NikonSi02xx.0x0083': '0 128 53 0 0 0 0 0 0 0 0 0 0 0 0 0 0 97 192 12 '
-                            '120 135 81 101 44 152 0 208 0 120 0 135 2 0 0 0 0 '
+                            '120 145 81 101 44 152 0 208 0 120 0 136 2 0 0 0 0 '
[...]
- 'Exif.NikonSi02xx.0x0159': '14 10 15 24 15 43 15 137 15 191 15 165 15 121 15 '
[...]
- 'Exif.NikonSi02xx.0x0257': '0 125 81 101 44 115 2 153 1 115 2 153 1 24 8 0 0 '
-                            '125 81 101 44 115 2 153 1 115 2 153 1 24 8',
- 'Exif.NikonSi02xx.0x027a': '101 44 108 10 76 1 1 0 0 24 16 0 0 0 0 0 1 0 0 0 '
[...]
```

For further info, `Exif.NikonSi02xx.0x027a` is a large binary blob, but even larger was ` Exif.Photo.MakerNote`, which was also not reported by `exiftool` (except for the version).

To make this short, can at least support for `WB_RGGBLevels` be added? I can provide info from `exivtool -U` if needed.

**Comment by boardhead** [comment]
Would it be possible to attach these samples?  At the very least, it would be useful to know what model camera this is.

- Phil

**Comment by StyXman** [comment]
Ah, sorry, forgot to mention. Nikon D7200. Samples are 13MiB each. 

https://dionecanali.hd.free.fr/~mdione/tmp/2023-08-08T19.18.31.jpg

https://dionecanali.hd.free.fr/~mdione/tmp/2023-08-08T19.18.41.jpg

**Comment by boardhead** [comment]
Sorry for the delay in dealing with this.

The RGGB values reported by the python lib start at offset 0x222 in the tag 0x0097 ColorBalance record with version number 0224.  I added the code to extract this and get the same values as reported by the python tool for the samples you provided, however I don't think these values are correct.  1) they jump around too much for these and other samples that I have, and 2) the G values are not the same, 3) I don't think the ColorBalance record is even being decrypted properly by ExifTool or the python utility -- when I fix the decryption the values at offset 0x222 change, and the same for both files.

I've looked at the ColorBalance information for your samples and the other D7200 samples I have, and I think I may have figured out where this is stored (at offset 0x080).  I'll run some more checks, and hopefully ExifTool 12.70 will have the ability to extract this.

- Phil
